### PR TITLE
Show MZ files in the PeakData advanced search results

### LINKED ISSUE #1262: Show mz files in the advanced search peakdata and peakgroups formats
### Inspiration

The mz files are searchable in the advanced search interface, but there is no column for them in the results.

### Description

Add a column for the mz files in the peakgroups and peakdata advanced search results.

### Alternatives

None

### Dependencies

None

### Comment

None

-----

### Assumptions

1. List of assumptions that the code will not explicitly address/check
2. E.g. We will assume input is correct (explaining why there is no validation)

### Limitations

1. A list of things this work will specifically not do
2. E.g. This feature will only handle the most frequent use case X

### Affected Components

- change: File path or DB table ...
- add: Environment variable or server setting
- delete: External executable or cron job

### Requirements

- [ ] 1. List of numbered conditions to be met for the feature
- [ ] 2. E.g. Every column/row must display a value, i.e. cannot be empty
- [ ] 3. Numbers for reference & checkboxes for progress tracking

#### Interface Change description

None provided

#### Code Change Description

None provided

#### Tests

- [ ] 1. A description of at least one test for each requirement above.
- [ ] 2. E.g. Test for req 2 that there's an exception when display value is ''
- [ ] 3. Numbers for reference & checkboxes for progress tracking

### PR DESCRIPTION
**Pull request by hepcat72**
## Summary Change Description

I also updated both the `PeakData` and `PeakGroups` tsv downloads.

## Affected Issues/Pull Requests

- Resolves #1262
- Merges into PR #1269

## Review Notes

See comments in-line.

## Checklist

This pull request will be merged once the following requirements are met.  The
author and/or reviewers should uncheck any unmet requirements:

- Review requirements
  - Minimum approvals: 1 
  - No changes requested
  - All blocking issues resolved by reviewers
  - Specific reviewers: @__add_username_here__
    
  - Review period: 2 days 
- Associated issue/pull request requirements:
  
  - [x] All requirements in affected issues marked "resolved" are satisfied
  - [x] All required pull requests are merged *(or none)*
- Basic requirements
  
  - [x] [All linters pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting)
  - [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)
  - [x] All conflicts resolved
- Overhead requirements
  
  - [x] [New/changed method tests implemented/updated *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
  - [ ] [Updated *Unreleased* section of `changelog.md` *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/changelog.md)
  - [x] [Migrations created & committed *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)

### CODE REVIEW

File: DataRepo/formats/peakdata_dataformat.py [hunk 1]
```diff
@@ -618,13 +618,14 @@ class PeakDataFormat(Format):
         },
         "MZFile": {
             "model": "ArchiveFile",
-            "path": "peak_group__msrun_sample__ms_data_file",
-            "reverse_path": "mz_to_msrunsamples__peak_groups__peak_data",
+            "path": "peak_group__msrun_sample__sample__msrun_samples__ms_data_file",
+            "reverse_path": "mz_to_msrunsamples__sample__msrun_samples__peak_groups__peak_data",
```
Comment (hepcat72) [inline DataRepo/formats/peakdata_dataformat.py:622]: C. Same as the peakgroups format edit. `MSRunSample` is traversed forwards and backwards to obtain all neighboring records from the sample in order to get to the `mzXML` files from the placeholder records that the `PeakGroup`s link to.

File: DataRepo/templates/DataRepo/search/results/peakdata.html [hunk 2]
```diff
@@ -59,7 +59,7 @@
         data-show-export="false"
         data-pagination="{% if mode == 'view' %}true{% else %}false{% endif %}">
 
-        <colgroup span="9" class="identdata"></colgroup>
+        <colgroup span="10" class="identdata"></colgroup>
```
Comment (hepcat72) [inline DataRepo/templates/DataRepo/search/results/peakdata.html:62]: C. I put the mz data files column in the first colgroup.

File: DataRepo/templates/DataRepo/search/downloads/peakgroups_colheads.tsv [hunk 3]
```diff
@@ -1 +1 @@
-Sample	Tissue	Time Collected (m)	Peak Group	Measured Compound(s)	Measured Compound Synonym(s)	Formula	Labeled Element	Total Abundance	Enrichment Fraction	Enrichment Abundance	Normalized Labeling	Peak Annotation Filename	Animal	Genotype	Body Weight (g)	Age (weeks)	Sex	Diet	Feeding Status	Treatment	Infusate	Tracer(s)	Tracer Compound(s)	Tracer Concentration(s) (mM)	Infusion Rate (ul/min/g)	Studies
```
Comment (hepcat72) [inline DataRepo/templates/DataRepo/search/downloads/peakgroups_colheads.tsv:1]: C. I included the peakgroup tsv edit that wasn't in the last PR.

File: DataRepo/templates/DataRepo/search/results/peakgroups.html [hunk 4]
```diff
@@ -121,11 +121,11 @@
                         {% endif %}
                     {% endwith %}
                 </th>
-                <th data-valign="top" data-sortable="{% if mode == 'view' %}true{% else %}false{% endif %}" data-sorter="alphanum" data-visible="{% get_template_cookie selfmt 'MZ_Filenames-data-visible' 'false' %}" data-field="MZ_Filenames" class="idgrp">
+                <th data-valign="top" data-sortable="false" data-visible="{% get_template_cookie selfmt 'MZ_Filenames-data-visible' 'false' %}" data-field="MZ_Filenames" class="idgrp">
```
Comment (hepcat72) [inline DataRepo/templates/DataRepo/search/results/peakgroups.html:124]: C. I realized I didn't need the sorting-related attributes.

File: DataRepo/templates/DataRepo/search/results/peakgroups.html [hunk 5]
```diff
@@ -335,9 +335,9 @@
                                 <!-- This block will execute when split_rows is False in the format definition.
                                 It displays all mzXML files associated with this peak group's sample in 1 table cell. -->
 
-                                <!-- Put displayed link text first for sorting -->
+                                <!-- Put displayed link text first for sorting (even though sorting is disabled until #1268 is finished) -->
                                 <div style="display:none;">
-                                    {% for msrsrec in pg.msrun_sample.sample.msrun_samples.all %}{% if not forloop.first %}{% endif %}{{ msrsrec.ms_data_file.filename }}{% endfor %}
+                                    {% for msrsrec in pg.msrun_sample.sample.msrun_samples.all %}{% if msrsrec.ms_data_file %}{{ msrsrec.ms_data_file.filename }}{% endif %}{% endfor %}
```
Comment (hepcat72) [inline DataRepo/templates/DataRepo/search/results/peakgroups.html:340]: C. I remoced the `forloop.first` conditional that was empty and added the conditional to skip the files that were `None`.

File: DataRepo/formats/peakdata_dataformat.py (hunk 1 above)
Comment (jcmatese) [reply DataRepo/formats/peakdata_dataformat.py:622]: Q: This query seems to be a little expansive (potentially including files potentially not directly linked to the peak_group)?  I don't have a problem with this, but we _might_ want to document it in a column note or tooltip?

**Review by jcmatese — APPROVED** [review]
This seems fine to me, but is there a separate PR to display associate datafiles on the sample display [which seemed more intuitive association (or at least additional), to me]?

File: DataRepo/formats/peakdata_dataformat.py (hunk 1 above)
Comment (hepcat72) [reply DataRepo/formats/peakdata_dataformat.py:622]: That's not a bad idea.  Even though the schema design prompting this query might change to eliminate this dual traversal of `MSRunSample`, we should lay bare the fact that this is a "bucket of files" associated with the sample and sequence.  I expect though that the schema changes will eliminate this need to traverse `MSRunSample` forward and backward.

- [ ] TODO: Add a tooltip to explain the relation between the files and the peakgroup.

**Pull request merged**